### PR TITLE
Add ddg2dnr as a sub-package

### PR DESCRIPTION
**Pull request by sammacbeth**
Merges https://github.com/duckduckgo/ddg2dnr as a subproject of this repo.

### CODE REVIEW

File: packages/ddg2dnr/tsconfig.json
Comment (shakyShane) [inline packages/ddg2dnr/tsconfig.json:1]: @sammacbeth  I would double-check that the Typescript setup is working as you expect. 

There tends to be 2 paths to go down during a migration like this

1) You can exclude type-checking the `packages/` part from the root tsconfig.json, and then ensure that `tsc` is also run for the workspace

or 

2) You configure the root-level tsconfig to also check child packages (via regular `include: ['...']`)

The reason I mention this, is that from the current PR I don't think the child `tsconfig.json` will work exactly as you imagine (eg: it's rules won't override the top level ones).

--

A 3rd option, which might not make sense here, is to use the recommended 'project references' setup, but that brings a whole heap of other issues too, like needing to compile `.d.ts` files for each project etc. (I've just beedn down this rabbit hole)

**Review by shakyShane — CHANGES_REQUESTED** [review]

**Review by kzar — APPROVED** [review]

File: packages/ddg2dnr/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "maxNodeModuleJsDepth": 0
+  },
+  "include": [
+      "*.js",
+      "**/*.js",
+      "node_modules/@duckduckgo/privacy-grade/src/classes/trackers.js"
+  ],
+  "types": [
+      "chrome"
+  ]
```
Comment (shakyShane) [inline packages/ddg2dnr/tsconfig.json:24]: This is incorrect (should be inside compilerOptions) - but actually it's not needed at all - TS will auto-include npm dependencies that are in `node_modules`
```suggestion
```

File: tsconfig.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
       "shared/js/background/events.js",
       "shared/js/background/onboarding.js",
       "shared/js/devtools/list-editor.js",
-      "shared/js/devtools/panel.js"
+      "shared/js/devtools/panel.js",
   ],
     "types": [
         "chrome"
```
Comment (shakyShane) [inline tsconfig.json:38]: as before, types: ["chrome"] can be removed

File: packages/ddg2dnr/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "checkJs": true,
+    "noEmit": true,
+    "allowSyntheticDefaultImports": true,
+    "resolveJsonModule": true,
+    "declaration": true,
+    "strict": true,
+    "noImplicitAny": false,
+    "strictNullChecks": true,
+    "maxNodeModuleJsDepth": 0
+  },
```
Comment (shakyShane) [inline packages/ddg2dnr/tsconfig.json:16]:
```suggestion
  "extends": "../../tsconfig.json",
```

**Review by shakyShane — CHANGES_REQUESTED** [review]

File: packages/ddg2dnr/tsconfig.json (hunk 3 above)
Comment (shakyShane) [reply packages/ddg2dnr/tsconfig.json:16]: @sammacbeth we can discourage config drift by sharing the compilerOptions from the root

File: packages/ddg2dnr/tsconfig.json (hunk 1 above)
Comment (shakyShane) [reply packages/ddg2dnr/tsconfig.json:24]: @sammacbeth Sorry if this wasn't part of the PR, but we might as well fix now :)

**Review by shakyShane — APPROVED** [review]

**Pull request merged**